phases/ephemeral/witx/typenames.witx

```diff
@@ -477,6 +481,31 @@
   )
 )
 
+;;; File permissions. This represents the permissions associated with a
+;;; file in a filesystem, and don't fully reflect all the conditions
+;;; which determine whether a given WASI program can access the file.
+(typename $permissions
+  (flags u8
+    ;;; For files, permission to read the file.
+    ;;; For directories, permission to do `readdir` within the directory.
+    $read
+
+    ;;; For files, permission to mutate the file.
+    ;;; For directories, permission to create, remove, and rename items
+    ;;; within the directory.
+    $write
+
+    ;;; For files, permission to "execute" the file, using whatever
+    ;;; concept of "executing" the host filesystem has.
+    ;;; This flag is not valid for directories.
+    $execute
+
+    ;;; For directories, permission to access files within the directory.
+    ;;; This flag is not valid for files.
```

Comment:
Why made this seprate from "read". It it useful to be able to list a directory but then not access the files within it?  Isn't the ability to access each file within a given directory determined by that file, and not by the directory?

Reply:
This comes from POSIX, which makes this distinction between the "r" and "x" bits on a directory. I agree though that this is pretty obscure. Shall we remove the search flag and just make read on directories imply search?

Reply:
If its rarely-used and POSIX-only then I think yes.  Like the U/G/O split its something we could add later maybe?

Reply:
Sounds good. I've now made this change.